server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/get/TransportGetRepositoriesAction.java

```diff
@@ -83,18 +84,33 @@ protected void masterOperation(
         if (repoNames.length == 0 || (repoNames.length == 1 && ("_all".equals(repoNames[0]) || "*".equals(repoNames[0])))) {
             return repositories.repositories();
         } else {
-            Set<String> repositoriesToGet = new LinkedHashSet<>(); // to keep insertion order
+            boolean seenWildcard = false;
```

Comment:
Do even need to have the `seenWildcard` variable? If we remove it, the tests passes. Is it used just as a perfomance optimization not to calculate negative patterns if there are no wildcards?

Reply:
Added tests now that would catch this :) It's needed to correctly handle the case where we request stuff that is named starting in a `-`.